src/openapi/transport/queue.js

```diff
@@ -173,17 +172,23 @@ TransportQueue.prototype.emptyQueue = function() {
 TransportQueue.prototype.runQueueItem = function(item) {
     this.transport[item.method]
         .apply(this.transport, item.args)
-        .then(function() {
-            item.resolve.apply(null, arguments);
+        .then((...args) => {
+            item.resolve(...args);
         },
-        function(result) {
-            if (this.transportAuth && result && result.status === 401) {
-                this.isQueueing = true;
+        (result, ...args) => {
+            if (this.authProvider && result && result.status === 401) {
                 this.addToQueue(item);
+                // if we are fetching a new token, wait
+                if (this.authProvider.isFetchingNewToken()) {

```

Comment:
This is another change - I suspect that if we get a 401 but we already have a new token, we wouldn't be fetchng and the rest call will be hanging until the next token comes in - that could be disastrous....

Reply:
we listen to a token received event and when we get it received, the queue is emptied.. so if a token refresh fails nothing happens.